AdyenNetworking/APIClient/HTTPDataResponse.swift

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright (c) 2023 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+import Foundation
+
+/// Protocol representing a HTTP response which contains raw response
+protocol AnyDataResponse: AnyHTTPResponse {
+    
+    /// Response data representing the raw response info in `Data`
+    var responseData: Data { get }
+}
+
+/// Struct conforming to `AnyDataResponse` which contains standard HTTP response info, including headers,
+/// status code and response body, as well as the raw `Data` response
+public struct HTTPDataResponse<R: Response>: AnyDataResponse {
+    
+    /// HTTP Headers.
+    public let headers: [String: String]
+    
+    /// HTTP Status Code.
+    public let statusCode: Int
+    
+    /// Response body
+    public let responseBody: R
+    
+    /// Raw data response
+    public let responseData: Data
```

Comment:
if you want the raw data response, then maybe AdyenNetworking SDK is not what you need, the whole idea of `AdyenNetworking` is to make the request and responses strongly typed and not to have to deal with low level http stuff like parsing json and handling status codes and what not and focus on business logic, if you now expose the raw data, then whats the point? :-)